### PR TITLE
Spreadsheet: Fix column index string to number conversion

### PR DESCRIPTION
**Pull request by Mandar12**
Issue: In Spreadsheet application, we have `convert_from_string` function to get column index from a string.
It was giving wrong output.
e.g. `convert_from_string("AB") == 52`. Correct output is 27.

Fixed calculation of index and added error handling.

Testing:
Added test case for `AK::String::bijective_base_from`
`convert_from_string` function is `static`, so manually tested output for few input values.

### CODE REVIEW

File: Userland/Applications/Spreadsheet/Spreadsheet.cpp [hunk 1]
```diff
@@ -82,24 +82,30 @@ size_t Sheet::add_row()
...
-        map = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
-
-    VERIFY(base >= 2 && base <= map.length());
-
     size_t value = 0;
-    for (size_t i = str.length(); i > 0; --i) {
-        auto digit_value = map.find(str[i - 1]).value_or(0);
+    const auto len = str.length();
+    for (auto i = 0u; i < len; i++) {
+        assert(str[i] >= 'A' && str[i] <= 'Z');
```
Comment (alimpfard) [inline Userland/Applications/Spreadsheet/Spreadsheet.cpp:90]: Please use `VERIFY()` instead of `assert()` here.

and I'm still against changing the behaviour of this function past fixing the actual bug.

Comment (Mandar12) [reply Userland/Applications/Spreadsheet/Spreadsheet.cpp:90]: I will add back `base` and `map`.

File: Userland/Applications/Spreadsheet/Spreadsheet.cpp [hunk 2]
```diff
@@ -208,7 +212,11 @@ Optional<Position> Sheet::parse_cell_name(const StringView& name) const
 
 Optional<size_t> Sheet::column_index(const StringView& column_name) const
```
Comment (TobyAsE) [inline Userland/Applications/Spreadsheet/Spreadsheet.cpp:213]: Why is this function even needed?
We know the index of a column to create the string representing its index in the first place. Whoever wants the index for a given column should be able to get it from the column directly or from whoever created the colns and keeps track of them. This route seems very round-about. 
But I guess this issue is out of scope of this pr...

Comment (alimpfard) [reply Userland/Applications/Spreadsheet/Spreadsheet.cpp:213]: That function is needed for the JS interop side of the story : )

**Comment by Mandar12** [comment]
CI failure seems unrelated to PR.
Please let me know if there are any more review comments.

**Comment by alimpfard** [comment]
LGTM, please just fixup the first commit's description as it is no longer entirely correct.

**Comment by Mandar12** [comment]
> LGTM, please just fixup the first commit's description as it is no longer entirely correct.

Thanks for pointing it out. I have changed the commit message.
